packages/core/mesh/teleport-extension-gossip/src/presence.ts

```diff
@@ -65,6 +61,18 @@ export class Presence {
       this._receiveAnnounces(message);
     });
 
+    // Remove peer state when connection is closed.
+    this._params.gossip.connectionClosed.on((peerId) => {
```

Comment:
Will route errors and unsubscribe automatically.

```suggestion
    this._params.gossip.connectionClosed.on(this._ctx, (peerId) => {
```

Although this will make this class non-reentrant (since the context would be different on the second open) so it's better to move this subscription into `_open` -- which is right anyway since you don't want the callback to be called on a closed Presence